test/Mono.Linker.Tests.Cases/DataFlow/DataFlowInAsyncCode.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+
+namespace Mono.Linker.Tests.Cases.DataFlow
+{
+	[SkipKeptItemsValidation]
+	[ExpectedNoWarnings]
+	public class DataFlowInAsyncCode
+	{
+		public static void Main ()
+		{
+			TestParameter (typeof (TestClass));
+			TestLocalVariable ();
+			TestGenericParameter<TestClass> ();
+		}
+
+		static async void TestParameter ([DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)] Type type)
```

Comment:
I checked and it looks like `async` and `iterator` methods each get an attribute, `Async/IteratorStateMachine` along with a `typeof` pointing to the definition. We could consider generalizing that in some way.